lib/phoenix/endpoint/watcher.ex

```diff
@@ -57,4 +59,39 @@ defmodule Phoenix.Endpoint.Watcher do
   end
 
   defp cd(opts), do: opts[:cd] || File.cwd!()
+
+  defp maybe_prefix_shell({:win32, _}, cmd, args, opts) do
+    # This lets us call batch files on Windows, which aren't proper executables and can't be simply
+    # passed to `System.cmd`.
+    case find_executable(cmd, opts[:cd]) do
+      nil ->
+        {cmd, args}
+
+      path ->
+        if String.ends_with?(path, [".cmd", ".bat"]) do
+          {"cmd", ["/c", Path.basename(path)] ++ args}
+        else
+          {cmd, args}
+        end
+    end
+  end
+
+  defp maybe_prefix_shell(_, cmd, args, _opts) do
+    {cmd, args}
+  end
+
+  defp find_executable(cmd, cd) do
+    path =
```

Comment:
`cmd` is either an absolute path or something in the OS path. So i think you need this:

```
if Path.type(cmd) == :relative do
  System.find_executable(cmd)
else
  cmd
end
```

No need to mangle `cd`. :)